src/main/kotlin/com/viartemev/thewhiterabbit/channel/Channels.kt

```diff
@@ -67,6 +86,18 @@ suspend fun Connection.confirmChannel(block: suspend ConfirmChannel.() -> Unit):
         .also { block(it) }
 }
 
+suspend fun Connection.txChannel(block: suspend TxChannel.() -> Unit): TxChannel {
+    var channel = Channels.localTxChannels[Thread.currentThread()]
+    if (channel == null || !channel.isOpen) {
+        channel = Channels.UnclosableTxChannel(createTxChannel())
+        localTxChannels.put(Thread.currentThread(), channel)
+    }
+    return channel
+        .also {
+            block(it)
+        }
+}
+
 suspend fun Connection.channel(block: suspend Channel.() -> Unit): Channel {
```

Comment:
Have started work on tx receive. @asm0dey, `Connection.channel` returns `Channels.UncloseableChannel(createConfirmChannel())`, i.e. the channel in confirm mode. And so does `Connection.confirmChannel` (in a different wrapper). Does it make sense?

Reply:
Of course that's an error! It doesn't make any sense and potentially breaks concurrency logics.
Thank you for ointing it!